sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -104,19 +103,24 @@ abstract class StringRegexExpression extends BinaryExpression
       spark.sql.parser.escapedStringLiterals	false
       > SELECT '%SystemDrive%\\Users\\John' _FUNC_ '\%SystemDrive\%\\\\Users%';
       true
+      > SELECT '%SystemDrive%/Users/John' _FUNC_ '/%SystemDrive/%//Users%' ESCAPE '/';
+      true
   """,
   note = """
     Use RLIKE to match with standard regular expressions.
   """,
   since = "1.0.0")
 // scalastyle:on line.contains.tab
-case class Like(left: Expression, right: Expression) extends StringRegexExpression {
+case class Like(left: Expression, right: Expression, escapeCharOpt: Option[Char] = None)
```

Comment:
How about just
```
escapeCharOpt: Char = '\\'
```

Reply:
It's OK too.

Reply:
yea this sounds better, to make the code simpler

Reply:
None indicates that `ESCAPE` is not specified, so that we can ignore it in `toString`. The existing code seems better.

Reply:
We can show `escapeChar` if it is not the default value. It is quite trivial for the method `toString`. 
But changing the data type of this parameter can simplify the rest of the code changes.

Reply:
If user uses SELECT ... LIKE ..., I think toString display `SELECT ... LIKE ...` is better than `SELECT ... LIKE ... ESCAPE \`.

Reply:
If the last parameter is '\' , then we can just ignore it in the `toString` method.

Reply:
If user uses `SELECT ... LIKE ... ESCAPE ''`, I think `toString` display `SELECT ... LIKE ... ESCAPE ` is better than `SELECT ... LIKE ...` .

Reply:
This is a good point. Users may specify the ESCAPE the same as the default value, and `toString` should display it.

However this is really a corner case. I'm fine either way. If this can simply code a lot, let's do it.

Reply:
OK.